public/app/core/components/Login/LoginServiceButtons.tsx

```diff
@@ -1,38 +1,42 @@
 import React from 'react';
 import config from 'app/core/config';
 
-const loginServices: () => LoginServices = () => ({
-  saml: {
-    enabled: config.samlEnabled,
-    name: 'SAML',
-    className: 'github',
-    icon: 'key',
-  },
-  google: {
-    enabled: config.oauth.google,
-    name: 'Google',
-  },
-  github: {
-    enabled: config.oauth.github,
-    name: 'GitHub',
-  },
-  gitlab: {
-    enabled: config.oauth.gitlab,
-    name: 'GitLab',
-  },
-  grafanacom: {
-    enabled: config.oauth.grafana_com,
-    name: 'Grafana.com',
-    hrefName: 'grafana_com',
-    icon: 'grafana_com',
-  },
-  oauth: {
-    enabled: config.oauth.generic_oauth,
-    name: config.oauth.generic_oauth ? config.oauth.generic_oauth.name : 'OAuth',
-    icon: 'sign-in',
-    hrefName: 'generic_oauth',
-  },
-});
+const loginServices: () => LoginServices = () => {
+  const oauthEnabled = !!config.oauth;
+
+  return {
+    saml: {
+      enabled: config.samlEnabled,
+      name: 'SAML',
+      className: 'github',
+      icon: 'key',
+    },
+    google: {
+      enabled: oauthEnabled && config.oauth.google,
```

Comment:
Nice catch with the oauthEnabled  - did you run into some problems with it?

Reply:
Yeah, this is what caused the bug. If oauth was undefined we got the error in the issue